File: 3.php
<?php

/**
 * 	Largest prime factor
 * 	The prime factors of 13195 are 5, 7, 13 and 29.
 *	What is the largest prime factor of the number 600851475143 ?
 *
 *	real	0m0.501s
 *	user	0m0.479s
 *	sys		0m0.016s
 */

$n = 600851475143;

function getFactors() {
	global $n;
	for ($i=2; $i<=ceil(sqrt($n)); $i++) {
		if ($n % $i === 0) $factors[] = $i;
	}
	return $factors;
}

function findLargestPrime() {
	global $factors;
	foreach ($factors as $f) {
		$x = array_pop($factors);
		if (isPrime($x)) return $x;
	}
}

function isPrime($x) {
	global $factors;
	foreach ($factors as $f) {
		if ($x % $f === 0) return false;
	}
	return true;
}

$factors = getFactors();
$largest = findLargestPrime();
print "Answer: $largest"; 
	
?>

File: 3.py
#	Largest prime factor
#	The prime factors of 13195 are 5, 7, 13 and 29.
#	What is the largest prime factor of the number 600851475143 ?

import math

n = 600851475143
factors = []

def get_factors(n):
	global factors
	for i in range(2, int(math.ceil(math.sqrt(n)))):
		if n % i == 0:
			factors.append(i)
	return factors

def find_largest_prime():
	global factors
	for f in factors:
		x = factors.pop()
		if is_prime(x):
			return x

def is_prime(x):
	for f in factors:
		if x % f == 0:
			return False
	return True

factors = get_factors(n)
largest = find_largest_prime()
print "Anser: %s" % largest